framework/wazuh/rbac/orm.py

```diff
@@ -2675,21 +2677,35 @@ def get_database_version(self, database: str) -> str:
         """
         return str(self.sessions[database].execute("pragma user_version").first()[0])
 
-    def insert_default_resources(self, database: str):
+    def insert_default_resources(self, database: str, old_database: str = None):
         """Insert default security resources into the given database.
 
         Parameters
         ----------
         database : str
             Name of the stored database.
+        old_database: str
+            Name of the previous database.
         """
         # Create default users if they don't exist yet
         with open(os.path.join(DEFAULT_RBAC_RESOURCES, "users.yaml"), 'r') as stream:
             default_users = yaml.safe_load(stream)
 
             with AuthenticationManager(self.sessions[database]) as auth:
                 for d_username, payload in default_users[next(iter(default_users))].items():
-                    auth.add_user(username=d_username, password=payload['password'], check_default=False)
+                    password = payload['password']
```

Comment:
Instead of migrating the current default users' passwords using the insert_default_resources method, I would have used a much cleaner approach modifying/refactoring the migrate_data function. It is important to maintain the goal and objective of each function as clear as possible.